metaflow/flowspec.py

```diff
@@ -465,6 +478,13 @@ def next(self, *dsts, **kwargs):
                 raise InvalidNextException(msg)
             funcs.append(name)
 
+        if cluster_size:
+            assert (
```

Comment:
we have avoided `assert` since `python -O` disables asserts. We can raise a proper Exception here